unikube/cli/console/deck.py

```diff
@@ -1,48 +1,55 @@
-from typing import Union
+from typing import Optional
+from uuid import UUID
 
 import unikube.cli.console as console
+from unikube.cache import UserIDs
 from unikube.cli.console.input import get_identifier_or_pass
 from unikube.context.helper import convert_deck_argument_to_uuid
-from unikube.graphql_utils import GraphQL
-
-
-def deck_list(ctx, organization_id: str = None, project_id: str = None) -> Union[None, str]:
-    # GraphQL
-    try:
-        graph_ql = GraphQL(authentication=ctx.auth)
-        data = graph_ql.query(
-            """
-            query($organization_id: UUID, $project_id: UUID) {
-                allDecks(organizationId: $organization_id, projectId: $project_id) {
-                    results {
-                        title
-                        id
-                        project {
-                            id
-                            title
-                            organization {
-                                id
-                            }
-                        }
-                    }
-                }
-            }
-            """,
-            query_variables={
-                "organization_id": organization_id,
-                "project_id": project_id,
-            },
-        )
-        deck_list = data["allDecks"]["results"]
-    except Exception as e:
-        console.debug(e)
-        console.exit_generic_error()
+
+
+def deck_list(ctx, organization_id: UUID = None, project_id: UUID = None) -> Optional[UUID]:
+    user_ids = UserIDs(id=ctx.user_id)
+    if not user_ids.deck:
+        user_ids.refresh()
```

Comment:
Just wondering - can we be sure that the user is authenticated here?

Reply:
No, I guess we cannot. However, the GraphQL class has a retry logic which tries to authenticate the user it the request fails.